src/jquery.multiselect.js

```diff
@@ -307,13 +307,16 @@
           });
 
           // Build the list section for this optgroup, complete w/ option inputs...
-          var $optionGroup = $( document.createElement('ul') )
-                .addClass('ui-multiselect-optgroup' 
-                          + (self.options.groupColumns ? ' ui-multiselect-columns' : '') 
-                          + (elem.className && ' ') + elem.className)
-                .append( $( document.createElement('a') ).text( elem.getAttribute('label') ), options);
-
-          list.push($optionGroup);
+          var $optGroupItem = $( document.createElement('li') )
+            .addClass('ui-multiselect-optgroup' 
+            + (self.options.groupColumns ? ' ui-multiselect-columns' : '') 
+            + (elem.className && ' ') + elem.className)
+            .append( $( document.createElement('a') )                  
+              .text( elem.getAttribute('label') ) );
+          var $optionGroup = $( document.createElement('ul') ).append(options)
+          $optGroupItem.append($optionGroup)
```

Comment:
A nitpick:  I would suggest breaking out the anchor to save an append call:

var $optGroupItem = $( document.createElement('li') )
           .addClass('ui-multiselect-optgroup' 
             + (self.options.groupColumns ? ' ui-multiselect-columns' : '') 
              + (elem.className && ' ') + elem.className);
var $optGroupAnchor =$( document.createElement('a') ).text( elem.getAttribute('label') );
var $optionGroup = $( document.createElement('ul') ).append(options);

$optGroupItem.append($optGroupAnchor, $optionGroup);